lib/active_record/virtual_attributes/virtual_fields.rb

```diff
@@ -191,7 +191,10 @@ def grouped_records(orig_association, records, polymorphic_parent)
               case association
               when Symbol, String
                 reflection = record.class._reflect_on_association(association)
-                next if polymorphic_parent && !reflection || !record.association(association).klass
+                if !reflection || !record.association(association).klass
```

Comment:
Based on the previous `if`, I feel like it is safer to just keep `polymorphic_parent` in it as is:

```ruby
if polymorphic_parent && !reflection || !record.association(association).klass
```

Since it expands to this:

```
if (polymorphic_parent && !reflection) || !record.association(association).klass
```

Where now, `!reflection` isn't dependent of `polymorphic_parent` being in place.

I might not matter, but I think for this being a minor update, I would rather have the least amount of code/logic change to get the desired effect.

Reply:
if we did not want to display a deprecation warning, then we could just modify the `if`.

but we have 3 cases:

1. association is found. include it.
2. association is not found but it has a polymorphic parent. just skip it.
3. association is not found and parent is not polymorphic. warn and skip (this changed)

We were following the rails 6.0 logic. now we are introducing leniency in case 3.

Reply:
> but we have 3 cases:

But the `if` isn't 3 cases.  None of those cases account for the `!reflection` portion of the statement, and it's original form, that is dependent on `polymorphic_parent`.

In the "new form", it is different, since `polymorphic_parent` now applies to both `!reflection` and the `!record.association(association).klass` case.

Reply:
DISREGARD!

I forgot that we were blowing up before, and now we are trying to not do that as part of this change.  Derp.

That said, do we eventually revert to the old if then?

Reply:
Probably best to have semantic errors throw errors.

do this as a todo?

Reply:
> Probably best to have semantic errors throw errors.

Huh?  I don't understand what this means.